### PR TITLE
Adding the SoftMax Function

### PR DESCRIPTION
**Pull request by RicardoRibeiroRodrigues**
## Summary of change
This pull request introduces an implementation of the SoftMax algorithm. The SoftMax function is often used as the last activation function of a neural network to normalize the output of a network to a probability distribution over predicted output classes.

## [Definition](https://en.wikipedia.org/wiki/Softmax_function)
The SoftMax function takes as input a vector z of K real numbers, and normalizes it into a probability distribution consisting of K probabilities proportional to the exponentials of the input numbers.

## Motivation
The SoftMax function is a critical component in many machine learning models, particularly in classification tasks within neural networks.

## Time Complexity
The time complexity for the SoftMax algorithm is `O(n)`, where n is the number of elements in the input array.

- [x] I have performed a self-review of my code
- [X] My code follows the style guidelines of this project
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Comments in areas I changed are up to date
- [X] I have added comments to hard-to-understand areas of my code
- [X] I have made corresponding changes to the README.md

### CODE REVIEW

**Review by siriak — APPROVED** [review]
Looks good, thanks!
